src/main/java/net/snowflake/client/core/SFBaseSession.java

```diff
@@ -392,21 +392,38 @@ public HttpClientSettingsKey getHttpClientKey() throws SnowflakeSQLException {
       boolean httpUseProxy = Boolean.parseBoolean(systemGetProperty("http.useProxy"));
       String httpProxyHost = systemGetProperty("http.proxyHost");
       String httpProxyPort = systemGetProperty("http.proxyPort");
+      String httpProxyUser =
+          (systemGetProperty("http.proxyUser") != null) ? systemGetProperty("http.proxyUser") : "";
+      String httpProxyPassword =
+          (systemGetProperty("http.proxyPassword") != null)
+              ? systemGetProperty("http.proxyPassword")
+              : "";
       String httpsProxyHost = systemGetProperty("https.proxyHost");
       String httpsProxyPort = systemGetProperty("https.proxyPort");
+      String httpsProxyUser =
+          (systemGetProperty("https.proxyUser") != null)
+              ? systemGetProperty("https.proxyUser")
+              : "";
+      String httpsProxyPassword =
+          (systemGetProperty("https.proxyPassword") != null)
```

Comment:
can we add a method with default value when and do not double the property name?

Reply:
Actually, it is redundant to check the return value here. Aligned it with the rest of the fields.